### ISSUE TITLE
Unable to save model when using Mirrored Strategy with multiple GPUs

### ISSUE THREAD

**Issue by richriley** [tensorflow/tensorflow#31131]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian GNU/Linux 9 (stretch) (Linux 4.9.0-9-amd64)
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v1.12.1-7260-gd4bc7d98c6 2.0.0-dev20190729
- Python version: 3.7.2
- GPU model and memory: 2 X Tesla V100 (24 gig RAM)

**Describe the current behavior**
An error is thrown when saving a keras model that has been compiled in using a Mirrored Strategy using more than one GPU.

**Describe the expected behavior**
Should be able to save and reload models that have been saved under Mirror Strategy.

**Code to reproduce the issue**

```import tensorflow as tf
from tensorflow.keras.applications.inception_v3 import InceptionV3

mirrored_strategy = tf.distribute.MirroredStrategy()


def get_model():
    with mirrored_strategy.scope():
        model = InceptionV3(input_shape=(299, 299, 3),
                                 include_top=True,
                                 weights='imagenet',
                                 pooling='avg')

        model.compile(loss='sparse_categorical_crossentropy',
                      optimizer=tf.keras.optimizers.Adam(),
                      metrics=['accuracy'])

    return model


model = get_model()
model.save("my_model")
```

**Other info / logs**
See attached.
[logs.txt](https://github.com/tensorflow/tensorflow/files/3442574/logs.txt)

**Comment by rishabhsahrawat** [comment]
You need to define the model function inside `strategy.scope()` or remove the function and just use 
`
         with mirrored_strategy.scope():

        model = InceptionV3(input_shape=(299, 299, 3),
                             include_top=True,
                             weights='imagenet',
                             pooling='avg')

        model.compile(loss='sparse_categorical_crossentropy',
                      optimizer=tf.keras.optimizers.Adam(),
                      metrics=['accuracy'])`

**Comment by oanush** [comment]
@richriley ,
Can you please let us know if solution provided by @rishabhsahrawat worked? Thanks!

**Comment by richriley** [comment]
It seems that my original code now works on 2.0.0-dev20190809. I can confirm that @rishabhsahrawat's example also works.

Thanks for your help.

**Comment by oanush** [comment]
Closing since the issue is resolved.Thanks!

**Comment by oanush** [comment]
Closing since the issue is resolved.Thanks!
